lib/ApparelSorter.js

```diff
@@ -213,25 +203,67 @@ function findRegexMatch(patterns, iteration, size) {
     }
     const sizeObj = {
         regex: "No Match", 
-        index: parseInt(size.replace(/[^\d.-]/g, '')), 
+        index: parseSizing(size), 
         size: size,
         normalized: size,
         sizeVal: extractSizeVal(size, size)
-    }
+    };
     return sizeObj;
 }
 
-const parseSizeVal = (size) => {
-    let sizeInt = parseInt(size) || 0;
-    const hasLengthParam = size.includes('Short') || size.includes('Long');
+const parseSizing = (size) => {
+    let sizeInt;
+    const isValid = (size) => isNaN(size) == false;
+    try {
+        // First, attempt to parse the size directly.
+        sizeInt = getNumericQuantity(size);
+
+        if (isValid(sizeInt)) {
+            return sizeInt;
+        }
+        // If not valid number, attempt to split the string on spaces and parse each (while filtering out words)
+        const allComponentPiecesSpaceDelimited = [
+            ...size
+                .split(' ')
+                .filter(
+                    val => new RegExp(/^[a-zA-Z]+$/).test(val) == false
+                )
+            ];
+
+        // Add the decimal values together
+        sizeInt = allComponentPiecesSpaceDelimited.reduce((accumulator, currValue) => {
+            const valueToAdd = getNumericQuantity(currValue);
+            return accumulator + valueToAdd;
+        }, 0);
+
+        if (isValid(sizeInt)) {
+            return sizeInt;
+        }
+
+    } catch (_) {
+        return parseInt(size) || 0
+    }
+    // If we haven't been able to parse anything out, fall back to parse int with str replacement (this will force the size to the end, unfortunately).
+    return parseInt(size.replace(/[^\d.-]/g, ''))
+}
+
+const applyWeightingToSizing = (size, sizeInt) => {
+    const hasLengthParam = (size.toLowerCase() || '').includes('short') || (size.toLowerCase() || '').includes('long');
 
     if (hasLengthParam) {
-        sizeInt = size.includes('Short') ? sizeInt : sizeInt + 0.5;
+        sizeInt = (size.toLowerCase() || '').includes('short') ? sizeInt - 0.005 : sizeInt + 0.005;
```

Comment:
So we'd 'penalize' `Short`, but `Regular` & `Long` (or any other lengths) would get an additional `0.005`? I think that's probably fine, otherwise we could have a lookup object and assign `0.000` to `Regular`.

Reply:
Basically! This is kind of a rudimentary logic based on what currently exists in SSC. I could seeing us doing some updating down the line where we have a list of sorting keywords that we either penalize or promote.